zerver/views/typing.py

```diff
@@ -45,6 +49,35 @@ def send_notification_backend(
         stream = access_stream_by_id(user_profile, stream_id)[0]
         access_stream_for_send_message(user_profile, stream, forwarder_user_profile=None)
         do_send_stream_typing_notification(user_profile, operator, stream, topic)
+
+    elif recipient_type_name == "message_edit":
+        if message_id is None:
+            raise JsonableError(_("Missing message_id"))
+
+        if stream_id is None:
+            raise JsonableError(_("Missing stream_id"))
+
+        if not user_profile.send_stream_typing_notifications:
+            raise JsonableError(_("User has disabled typing notifications for stream messages"))
+
+        stream = access_stream_by_id(user_profile, stream_id)[0]
+        access_stream_for_send_message(user_profile, stream, forwarder_user_profile=None)
+        do_send_message_edit_typing_notification(user_profile, stream_id, message_id, operator)
```

Comment:
Why is a stream ID required for the message_edit case? I would think we could just have the client sent the message ID itself, and the server can figure out whether it's a stream or direct message, and just look up the message itself (and the user's access to and sender status of it) and simplify this code path greatly.

Reply:
I added stream ID so as to send the typing notification event only to the users subscribed to the stream in which the message is present. (I couldn't find any method that obtains the `stream_id` from the `message_id`)

Reply:
Updated!